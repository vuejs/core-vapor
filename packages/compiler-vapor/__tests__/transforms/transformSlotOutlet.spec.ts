import { NodeTypes } from '@vue/compiler-core'
import {
  IRNodeTypes,
  transformChildren,
  transformElement,
  transformSlotOutlet,
  transformText,
  transformVBind,
  transformVOn,
} from '../../src'
import { makeCompile } from './_utils'

const compileWithSlotsOutlet = makeCompile({
  nodeTransforms: [
    transformText,
    transformSlotOutlet,
    transformElement,
    transformChildren,
  ],
  directiveTransforms: {
    bind: transformVBind,
    on: transformVOn,
  },
})

describe('compiler: transform <slot> outlets', () => {
  test('default slot outlet', () => {
    const { ir, code, vaporHelpers } = compileWithSlotsOutlet(`<slot />`)
    expect(code).toMatchSnapshot()
    expect(vaporHelpers).toContain('createSlot')
    expect(ir.block.effect).toEqual([])
    expect(ir.block.operation).toMatchObject([
      {
        type: IRNodeTypes.SLOT_OUTLET_NODE,
        id: 0,
        name: {
          type: NodeTypes.SIMPLE_EXPRESSION,
          content: 'default',
          isStatic: true,
        },
        props: [],
        fallback: undefined,
      },
    ])
  })

  test('statically named slot outlet', () => {
    const { ir, code } = compileWithSlotsOutlet(`<slot name="foo" />`)
    expect(code).toMatchSnapshot()
    expect(ir.block.operation).toMatchObject([
      {
        type: IRNodeTypes.SLOT_OUTLET_NODE,
        id: 0,
        name: {
          type: NodeTypes.SIMPLE_EXPRESSION,
          content: 'foo',
        },
      },
    ])
  })
})
