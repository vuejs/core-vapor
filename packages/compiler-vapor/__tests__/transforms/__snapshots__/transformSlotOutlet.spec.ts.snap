// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`compiler: transform <slot> outlets > default slot outlet 1`] = `
"import { createSlot as _createSlot } from 'vue/vapor';

export function render(_ctx) {
  const n0 = _createSlot("default")
  return n0
}"
`;

exports[`compiler: transform <slot> outlets > statically named slot outlet 1`] = `
"import { createSlot as _createSlot } from 'vue/vapor';

export function render(_ctx) {
  const n0 = _createSlot("foo")
  return n0
}"
`;
